core/manifest/ClassLoader.php

```diff
@@ -101,6 +101,7 @@ public function getItemPath($class) {
 	 * @return bool
 	 */
 	public function classExists($class) {
+		Deprecation::notice('3.3', 'Not necessary anymore as this function is inlined in \'ClassInfo::exists\' for speed and that\'s the only place where it is used in core.');
```

Comment:
Please set deprecation level to 4.0 (the version this will be removed in).

The description should just be "Use ClassInfo::exists"; You don't need to include the justification in the notice. Maybe that's better as a comment?

I've also noticed here that this mechanism does NOT work well with psr-4 autoloading. Is that going to be a problem? If a class isn't already loaded, and requires psr-4 to load, then it will incorrectly report false.